core/shared/src/main/scala/scalaz/zio/Duration.scala

```diff
@@ -0,0 +1,29 @@
+package scalaz.zio
+
+sealed trait Duration
+
+object Duration {
+
+  class Finite(duration: scala.concurrent.duration.FiniteDuration) extends Duration {
+    
+    def toMillis: Long = duration.toMillis
+
+    def toNanos: Long = duration.toNanos
+
+    def *(factor: Double): Duration = Duration.fromNanos((this.toNanos * factor).round)
+  }
+
+  case class Infinite() extends Duration
+
+  def fromNanos(nanos: Long): Finite = new Finite(scala.concurrent.duration.Duration.fromNanos(nanos))
+
+  def fromDuration(duration: scala.concurrent.duration.Duration): Option[Duration] =
+    duration match {
+      case finiteDuration: scala.concurrent.duration.FiniteDuration => Some(new Finite(finiteDuration))
```

Comment:
We should treat `< 0` as `0` (in case the Scala duration has a length of < 0).

Reply:
sorry to jump in, but I thought you decided that `scala.Duration` < 0 should be turned into `zio.Duration.Infinity`. is this a change of plans?

Reply:
Sorry, I mean only that the following should hold:

```scala
fromDuration(d) == fromNanos(d.toNanos)
```

For negative numbers, they should both go to `Infinity`.